drivers/i2c/i2c_nrfx_twim.c

```diff
@@ -297,40 +297,26 @@ static const struct i2c_driver_api i2c_nrfx_twim_driver_api = {
 	.recover_bus = i2c_nrfx_twim_recover_bus,
 };
 
-#ifdef CONFIG_PM_DEVICE
 static int twim_nrfx_pm_action(const struct device *dev,
 			       enum pm_device_action action)
 {
 	const struct i2c_nrfx_twim_config *dev_config = dev->config;
-	int ret = 0;
 
 	switch (action) {
 	case PM_DEVICE_ACTION_RESUME:
```

Comment:
`pm_device_driver_init` calls `PM_DEVICE_ACTION_TURN_ON` which will now end up in default case and it will return `-ENOTSUP`.

Reply:
The first commit in this PR updates pm to accept -ENOTSUP as a valid response to an action :) -ENOTSUP is correct since ON does not mean anything for the TWIM :)